### PR TITLE
Add INGP Docs

### PR DESCRIPTION
**Pull request by ayaanzhaque**
Let me know if anything looks inaccurate or if more information should be added

### CODE REVIEW

File: docs/nerfology/methods/instant_ngp.md [hunk 1]
```diff
@@ -8,13 +8,58 @@
...
+Nerfstudio uses the [tinycudann](https://github.com/NVlabs/tiny-cuda-nn) framework to utilize the fully-fused neural networks.
+
+The speedups at each level are multiplicative. With all their improvements, Instant-NGP reaches speedups of 1000x, which enable training NeRF scenes in a matter of seconds!
+
+### Multi-Resolution Hash Encoding
+
+```{image} imgs/ingp/hash_figure.png
+:align: center
+```
+The largest contribution of Instant-NGP is the multi-resolution hash encoding. In the traditional NeRF pipelines, input coordinates are mapped to a higher dimensional space using a positional encoding function, which is described [here](https://docs.nerf.studio/en/latest/nerfology/methods/nerf.html#positional-encoding). Instant-NGP proposes a trainable hash-based encoding. The idea is to map coordinates to trainable feature vectors which can be optimized in the standard flow of NeRF training.
```
Comment (tancik) [inline docs/nerfology/methods/instant_ngp.md:49]: `The largest contribution` -> `One contribution`

**Review by tancik — COMMENTED** [review]
This is a great start, thanks @ayaanzhaque! One component that is missing how sampling is done using an occupancy grid. We also should add a section on how the nerfstudio version differs from the paper implementation, perhaps @liruilong940607 has some comments along that front.

**Comment by ayaanzhaque** [comment]
I added the details about the occupancy grid. I'm not so sure how their implementation differs from the nerfstudio implementation, will ask ruilong

**Comment by liruilong940607** [comment]
The one in the master branch is with `nerfacc==0.3.5` which uses single res occ grid with contraction function. `nerfacc==0.5.x` supports multires occ grid, and I was doing this upgrade in #1705. Maybe hold on this doc until that upgrade is done? I think I'll separate this nerfacc upgrade into a new PR in this weekend.

**Comment by liruilong940607** [comment]
> This is a great start, thanks @ayaanzhaque! One component that is missing how sampling is done using an occupancy grid. We also should add a section on how the nerfstudio version differs from the paper implementation, perhaps @liruilong940607 has some comments along that front.

Maybe something like [this](https://www.nerfacc.com/en/stable/methodology/sampling.html#occupancy-grid-estimator) or just simply refer to that link?

**Comment by ayaanzhaque** [comment]
@liruilong940607 @tancik I added the description for the Instant-NGP occupancy grid and put a link to the Nerfacc sampling algorithm description. I think that should be good for now, but we can publish this docs once the new nerfacc changed are pushed

**Comment by ayaanzhaque** [comment]
Thoughts on merging this? What do we have to add?

**Comment by liruilong940607** [comment]
I think the writings are good. Would be nice to add two things: 
1/ Put a pointer to the `nerfacto` model in the end of doc, saying the fully-fused MLP and hash encoder are also equipped in the `nerfacto` model, inspired by Instant-NGP.
2/ Add a disclaimer that saying this implementation covers the major ideas from NGP, but isn't strictly following every details. Known differences include learning rate schedule, hyper-parameters for sampling and how camera gradients are calculated if enabled.

**Comment by ayaanzhaque** [comment]
@liruilong940607 @tancik just added those changes, lmk if we are good to merge!

**Review by tancik — APPROVED** [review]
Good on my end, I'll let @liruilong940607 also approve.

**Comment by liruilong940607** [comment]
Thanks for the update. LGTM!

**Comment by wanger-666** [comment]
_For other inputs such as the viewing direction, Instant-NGP encodes them using the traditional positional encoding from the original NeRF paper._It should be a spherical harmonic encoding
